src/Semaphore/SemaphorePrediction.ts

```diff
@@ -0,0 +1,19 @@
+import EncodingEntry from '../Common/EncodingEntry';
+import SemaphoreData from './SemaphoreData';
+import SemaphoreDirection from './SemaphoreDirection';
+import SemaphoreEncoding from './SemaphoreEncoding';
+
+class SemaphorePrediction {
```

Comment:
So I think this is completely unnecessary.  All classes already support prediction, simply do:

```
const ch = new SemaphoreCharacter();
const matches = ch.getPotentialMatches();
```

I'd rather continue that approach than introduce this side concept.